gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemTestBase.java

```diff
@@ -539,4 +542,94 @@ public void provideCoverageForUnmodifiedMethods() throws IOException {
       }
     }
   }
+
+  @Test
+  public void testGetFileStatusWithHint() throws Exception {
+    Path hadoopPath = ghfsHelper.castAsHadoopPath(getTempFilePath());
+    ghfsHelper.writeFile(hadoopPath, UUID.randomUUID().toString(), 1, /* overwrite= */ false);
+
+    // Running a few times to have a high probability of at least one eager operation finishing
+    int numTimes = 5;
+    Configuration configuration = new Configuration();
+    configuration.set(GoogleHadoopFileSystem.GETFILESTATUS_FILETYPE_HINT, "file");
+
+    invokeGetFileStatusWithHint(hadoopPath, numTimes, configuration);
+    assertThat(getStatisticValue(GCS_LIST_FILE_REQUEST)).isEqualTo(0);
+    assertThat(getStatisticValue(GCS_API_REQUEST_COUNT)).isEqualTo(numTimes);
+
+    invokeGetFileStatusWithHint(hadoopPath.getParent(), numTimes, configuration);
+    assertThat(getStatisticValue(GCS_LIST_FILE_REQUEST)).isEqualTo(numTimes);
+    assertThat(getStatisticValue(GCS_API_REQUEST_COUNT)).isEqualTo(numTimes * 2);
+
+    invokeGetFileStatusWithHint(hadoopPath, numTimes, new Configuration());
+    // Update happening in a different thread. Wait a few ms.
+    waitForUpdation(GCS_LIST_FILE_REQUEST, numTimes);
+
+    // Some of the LIST_FILE requests might have been cancelled before completion.
+    assertThat(getStatisticValue(GCS_LIST_FILE_REQUEST)).isGreaterThan(0);
+    assertThat(getStatisticValue(GCS_LIST_FILE_REQUEST)).isLessThan(numTimes + 1);
+    assertThat(getStatisticValue(GCS_API_REQUEST_COUNT)).isGreaterThan(numTimes);
+    assertThat(getStatisticValue(GCS_API_REQUEST_COUNT)).isLessThan(numTimes * 2 + 1);
+
+    ghfsHelper.delete(hadoopPath.toUri(), true);
+  }
+
+  private void waitForUpdation(GoogleCloudStorageStatistics stat, int value)
+      throws InterruptedException {
+    for (int i = 0; i < 10; i++) {
+      if (getStatisticValue(stat) >= value) {
+        return;
+      }
+
+      Thread.sleep(1);
```

Comment:
this will still result in test flakiness?

Reply:
I assume the other thread should catch up in 10 ms. If not, it is safe to increase the busy wait loop count.